mediachain/transactor/blockchain_follower.py

```diff
@@ -65,48 +68,82 @@ def cancel(self):
         self._cancelled = True
         self.block_ref_queue.put('__abort__')
         self.incoming_event_queue.put('__abort__')
+        # setting these events will cause the __abort__ messages to be
+        # pulled from the queues.
+        self.catchup_begin.set()
+        self.catchup_complete.set()
+
+    def _clear_queues(self):
+        with self.block_ref_queue.mutex:
```

Comment:
hmm what happens if something is inserted into a previously cleared queue in the middle of this func?